addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/LEDStateDTO.java

```diff
@@ -70,18 +75,18 @@ public static LEDStateDTO valueOf(int state, int program, int programSpeed, int
         OnOffType power = (state & 0x01) != 0 ? OnOffType.ON : OnOffType.OFF;
         float[] hsv = new float[3];
         Color.RGBtoHSB(red, green, blue, hsv);
-        long hue = (long) (hsv[0] * 360);
-        int saturation = (int) (hsv[1] * 100);
-        int brightness = (int) (hsv[2] * 100);
+        BigDecimal hue = BigDecimal.valueOf(hsv[0] * 360);
```

Comment:
My point is:

> Fix for integer division calculation in LEDStateDTO and changed them to using BigDecimal.

That the statement of your commit message is not correct, instead you switched to float calculations and store them in a BigDecimal, that does not seem like a step forward to me. So let's take a step back and find out what the problem is which we try to solve and how we are getting there.

I assume hue stores an integer value between 0 and 360 and the others an integer value between 0 and 100, the only potential problem I see is that it does not do rounding but always uses `flooring` But even the impact of that seems minimal to me.